arctic/decorators.py

```diff
@@ -47,7 +48,7 @@ def f_retry(*args, **kwargs):
             while True:
                 try:
                     return f(*args, **kwargs)
-                except (DuplicateKeyError, ServerSelectionTimeoutError) as e:
+                except (DuplicateKeyError, ServerSelectionTimeoutError, BulkWriteError) as e:
```

Comment:
Out of interest, what was the underlying error? Should we inspect `e.details` and handle the different cases?

Reply:
Agreed, i think this change is sensible.